examples/01_basic_ping_bot/src/main.rs

```diff
@@ -59,54 +58,74 @@ impl From<TungsteniteError> for Error {
 }
 
 fn main() {
-    tokio::run(try_main().map_err(|why| {
-        println!("Error running shard: {:?}", why);
-    }).boxed().compat(TokioDefaultSpawner));
+    tokio::run(main_future().map_err(|reason| {
+        eprintln!("Error when running shard {:?}:", reason);
+    }).boxed().compat());
 }
 
-async fn try_main() -> Result<(), Error> {
+async fn main_future() -> Result<(), Error> {
     env_logger::init();
-
     // Configure the client with your Discord bot token in the environment.
-    let token = env::var("DISCORD_TOKEN").expect("Expected a token in the environment");
-
-    // Create a new shard, specifying the token, the ID of the shard (0 of 1),
-    // and a handle to the event loop
-    let mut shard = await!(Shard::new(token, [0, 1]).compat())?;
-    println!("Connected shard");
-    let mut messages = shard.messages().compat();
+    let token = env::var("DISCORD_TOKEN").expect("No token was provided for the bot.");
+    // Create new shard.
+    let mut shard = await!(Shard::new(token, [0, 1]).compat()).unwrap();
+    let mut messages = shard.messages().expect("No shard messages found.").compat();
+    println!("Shard is connected.");
 
     loop {
         // Loop over websocket messages.
         let result: Result<_, Error> = try {
-            println!("Getting message");
             let message = await!(messages.next())??;
-
-            // Parse the websocket message into a serenity GatewayEvent.
+            println!("Receiving shard message {:?}", message);
+            // Parse websocket event to Serenity Gateway Event.
             let event = shard.parse(&message)?;
-
-            // Have the shard process the WebSocket event, in case it needs
-            // to mutate its state, send a packet, etc.
-            //
-            // This can give back a future in the event something needs to
-            // be done, such as waiting for a reconnection.
-            if let Some(future) = shard.process(&event) {
-                await!(future.compat())?;
-            }
-
-            // Now you can do whatever you want with the event.
+            //Process websocket event.
+            let process = shard.process(&event);
+            // Handle websocket actions, such as identifying and auto-reconnecting.
```

Comment:
Keep the empty line before the comment. Serenity style is to have an empty line before blocks and their comments.